pkg/storage/swift/swift.go

```diff
@@ -1,84 +1,223 @@
+		d.Config.TenantID = replaceEmpty(d.Config.TenantID, cloud.AuthInfo.ProjectID)
+		d.Config.Domain = replaceEmpty(d.Config.Domain, cloud.AuthInfo.DomainName)
+		d.Config.DomainID = replaceEmpty(d.Config.DomainID, cloud.AuthInfo.DomainID)
+		d.Config.Domain = replaceEmpty(d.Config.Domain, cloud.AuthInfo.UserDomainName)
+		d.Config.DomainID = replaceEmpty(d.Config.DomainID, cloud.AuthInfo.UserDomainID)
+		d.Config.RegionName = replaceEmpty(d.Config.RegionName, cloud.RegionName)
+
+		username = cloud.AuthInfo.Username
+		password = cloud.AuthInfo.Password
+	} else {
+		username = cfg.Storage.Swift.Username
+		password = cfg.Storage.Swift.Password
+	}
+
+	opts := &gophercloud.AuthOptions{
+		IdentityEndpoint: d.Config.AuthURL,
+		Username:         username,
+		Password:         password,
+		DomainID:         d.Config.DomainID,
+		DomainName:       d.Config.Domain,
+		TenantID:         d.Config.TenantID,
+		TenantName:       d.Config.Tenant,
+	}
+
+	provider, err := openstack.AuthenticatedClient(*opts)
+	if err != nil {
+		return nil, err
+	}
+
+	endpointOpts := gophercloud.EndpointOpts{
+		Region: d.Config.RegionName,
+	}
+
+	client, err := openstack.NewContainerV1(provider, endpointOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+// NewDriver creates new Swift driver for the Image Registry
 func NewDriver(c *imageregistryv1.ImageRegistryConfigStorageSwift, listers *regopclient.Listers) *driver {
 	return &driver{
 		Config:  c,
 		Listers: listers,
 	}
 }
 
+// Secrets returns a map of the storage access secrets
 func (d *driver) Secrets() (map[string]string, error) {
+	cfg, err := clusterconfig.GetSwiftConfig(d.Listers)
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.Storage.Swift.Cloud == nil {
+		return map[string]string{
+			"REGISTRY_STORAGE_SWIFT_USERNAME": cfg.Storage.Swift.Username,
+			"REGISTRY_STORAGE_SWIFT_PASSWORD": cfg.Storage.Swift.Password,
+		}, nil
+	}
+
 	return nil, nil
 }
 
 func (d *driver) ConfigEnv() (envs []corev1.EnvVar, err error) {
-	envs = append(envs,
-		corev1.EnvVar{Name: "REGISTRY_STORAGE", Value: "swift"},
-		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_AUTHURL", Value: d.Config.AuthURL},
-		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_CONTAINER", Value: d.Config.Container},
-		corev1.EnvVar{
-			Name: "REGISTRY_STORAGE_SWIFT_USERNAME",
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: imageregistryv1.ImageRegistryPrivateConfiguration,
-					},
-					Key: "REGISTRY_STORAGE_SWIFT_USERNAME",
-				},
-			},
-		},
-		corev1.EnvVar{
-			Name: "REGISTRY_STORAGE_SWIFT_PASSWORD",
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: imageregistryv1.ImageRegistryPrivateConfiguration,
-					},
-					Key: "REGISTRY_STORAGE_SWIFT_PASSWORD",
-				},
-			},
-		},
-	)
```

Comment:
Restore this - we need to read the swift usernames and passwords from a secret.

Review @dmage 's doc on the credential flow - we want to apply the same pattern for OpenStack: https://github.com/openshift/cluster-image-registry-operator/blob/master/docs/credentials-flow.md

Reply:
Done